In [1]:
#Importing needed libraries
import numpy as np
import pandas as pd
import seaborn as sns
import scipy.stats as stats
from scipy.stats import f_oneway

In [6]:
#Data reading
data = pd.read_csv("Advertising.csv")

In [8]:
print(data)

     Unnamed: 0     TV  Radio  Newspaper  Sales
0             1  230.1   37.8       69.2   22.1
1             2   44.5   39.3       45.1   10.4
2             3   17.2   45.9       69.3    9.3
3             4  151.5   41.3       58.5   18.5
4             5  180.8   10.8       58.4   12.9
..          ...    ...    ...        ...    ...
195         196   38.2    3.7       13.8    7.6
196         197   94.2    4.9        8.1    9.7
197         198  177.0    9.3        6.4   12.8
198         199  283.6   42.0       66.2   25.5
199         200  232.1    8.6        8.7   13.4

[200 rows x 5 columns]


In [25]:
#Dropping unwanted columns from data set
df = data.drop(["Unnamed: 0"], axis = 1)
df

TV  Radio  Newspaper  Sales
0    230.1   37.8       69.2   22.1
1     44.5   39.3       45.1   10.4
2     17.2   45.9       69.3    9.3
3    151.5   41.3       58.5   18.5
4    180.8   10.8       58.4   12.9
..     ...    ...        ...    ...
195   38.2    3.7       13.8    7.6
196   94.2    4.9        8.1    9.7
197  177.0    9.3        6.4   12.8
198  283.6   42.0       66.2   25.5
199  232.1    8.6        8.7   13.4

[200 rows x 4 columns]

## 1) Explaining data

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   TV         200 non-null    float64
 1   Radio      200 non-null    float64
 2   Newspaper  200 non-null    float64
 3   Sales      200 non-null    float64
dtypes: float64(4)
memory usage: 6.4 KB


In [16]:
df.shape

(200, 4)

Data has 200 rows and 4 columns

In [17]:
df.head()

TV  Radio  Newspaper  Sales
0  230.1   37.8       69.2   22.1
1   44.5   39.3       45.1   10.4
2   17.2   45.9       69.3    9.3
3  151.5   41.3       58.5   18.5
4  180.8   10.8       58.4   12.9

## 2) Summary statistics

In [18]:
df.describe()

TV       Radio   Newspaper       Sales
count  200.000000  200.000000  200.000000  200.000000
mean   147.042500   23.264000   30.554000   14.022500
std     85.854236   14.846809   21.778621    5.217457
min      0.700000    0.000000    0.300000    1.600000
25%     74.375000    9.975000   12.750000   10.375000
50%    149.750000   22.900000   25.750000   12.900000
75%    218.825000   36.525000   45.100000   17.400000
max    296.400000   49.600000  114.000000   27.000000

Insights:
a) Highest amount is spent on TV based advertisement.
b) Lowest amount is spent on Radio based advertisement.
c) Mean amount spent on TV, Radio, Newspaper advertisements are 147.042500, 23.264000, 30.554000 resp.
d) Maximum amount spent on TV, Radio, Newspaper advertisements are 296.4, 49.6, 114 resp.

## 3) Checking whether the same amount was spent for the three advertisements (TV , Radio and Newspaper).

NULL HYPOTHESIS (Ho): The Amount spent on the 3 modes of Advertisement are the same.

ALTERNATE HYPOTHESIS (Ha): Atleast one of the amount spent shows significance difference compared to others.

In [24]:
F_statistic, pvalue  = f_oneway(df['TV'], df['Radio'],df['Newspaper'])
print(f"The calculated F statistic is : \t{F_statistic:.2f}")
print(f"\nThe P-value is :  \t{pvalue:.3f}")

alpha = 0.05

if pvalue < alpha:
    print("\nSince the P-value < 0.05, we reject the Ho")
elif pvalue > alpha:
    print("\nSince the P-value > 0.05, we accept the Ho")

The calculated F statistic is : 	358.85

The P-value is :  	0.000

Since the P-value < 0.05, we reject the Ho


Inference: The amount spent on 3 modes of advertisement are not same.